lib/ClientImpl.cc

```diff
@@ -244,6 +245,34 @@ void ClientImpl::createReaderAsync(const std::string& topic, const MessageId& st
                   std::placeholders::_2, topicName, msgId, conf, callback));
 }
 
+void ClientImpl::createTableViewAsync(const std::string& topic, const TableViewConfiguration& conf,
+                                      TableViewCallback callback) {
+    TopicNamePtr topicName;
+    {
+        Lock lock(mutex_);
+        if (state_ != Open) {
+            lock.unlock();
+            callback(ResultAlreadyClosed, TableView());
+            return;
+        } else if (!(topicName = TopicName::get(topic))) {
+            lock.unlock();
+            callback(ResultInvalidTopicName, TableView());
+            return;
+        }
+    }
+
+    TableViewImplPtr tableViewPtr =
+        std::make_shared<TableViewImpl>(shared_from_this(), topicName->toString(), conf);
+    auto self = shared_from_this();
+    tableViewPtr->start().addListener([callback, self](Result result, TableViewImplPtr tableViewImplPtr) {
```

Comment:
```suggestion
    tableViewPtr->start().addListener([callback](Result result, TableViewImplPtr tableViewImplPtr) {
```

We don't need to capture `self` because there is no field like `producers_` and `consumers_` in `ClientImpl`